Code/Framework/AzFramework/AzFramework/Entity/GameEntityContextComponent.cpp

```diff
@@ -374,4 +733,255 @@ namespace AzFramework
         AZ::ComponentApplicationBus::BroadcastResult(entityName, &AZ::ComponentApplicationBus::Events::GetEntityName, id);
         return entityName;
     }
+
+    
+    
+    //=========================================================================
+    // AddEntityToParentChildTree
+    //=========================================================================
+    void GameEntityContextComponent::AddEntityToParentChildTree(AZ::Entity* entity)
+    {
+        if(!entity) { return; }
+        
+        //Get Parent
+        AZ::EntityId parentId; // default = null root
+        if (auto* tc = entity->FindComponent<AzFramework::TransformComponent>())
+        {
+            parentId = tc->GetParentId(); // works without bus
+        }
+        
+        //Add parent to Child -> Parent tree.
+        const AZ::EntityId id = entity->GetId();
+        parentOf[id] = parentId;
+
+        //Add child to Parent -> [Children..] tree
+        //Returns a ref to the parent->vector<AZ::EntityId>
+        auto& siblings = EnsureChildList(childrenByParentTree, parentId);
+
+        // Check for duplicates. Then adds child to vector.
+        if (AZStd::find(siblings.begin(), siblings.end(), id) == siblings.end())
+        {
+            siblings.push_back(id);
+        }
+
+        // Connect to TransformNotificationBus for this entity.
+        if (!AZ::TransformNotificationBus::MultiHandler::BusIsConnectedId(id))
+        {
+            AZ::TransformNotificationBus::MultiHandler::BusConnect(id);
+        }
```

Comment:
This would imply that the game entity context listens to every transform bus message, for every entity involved.   This is worriesome from a perf perspecitve.    Perhaps we can make a broadcast bus, it connects to just once, that transform broadcasts to?  it probably only needs to know about heiarchy changes